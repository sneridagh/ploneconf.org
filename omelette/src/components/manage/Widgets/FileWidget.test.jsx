import React from 'react';
import { Provider } from 'react-intl-redux';
import { render, waitFor } from '@testing-library/react';
import configureStore from 'redux-mock-store';

import FileWidget from './FileWidget';

const mockStore = configureStore();

test('renders a file widget component', async () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const { container } = render(
    <Provider store={store}>
      <FileWidget id="my-field" title="My field" onChange={() => {}} />,
    </Provider>,
  );

  await waitFor(() => {});
  expect(container).toMatchSnapshot();
});
