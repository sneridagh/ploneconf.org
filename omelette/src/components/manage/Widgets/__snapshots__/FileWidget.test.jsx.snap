// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a file widget component 1`] = `
<div>
  <div
    class="inline field"
  >
    <div
      class="ui grid"
    >
      <div
        class="stretched row"
      >
        <div
          class="four wide column"
        >
          <div
            class="wrapper"
          >
            <label
              for="field-my-field"
            >
              My field
            </label>
          </div>
        </div>
        <div
          class="eight wide column"
        >
          <div
            class="file-widget-dropzone"
            tabindex="0"
          >
            <div
              class="dropzone-placeholder"
            >
              <p
                class="dropzone-text"
              >
                Drop file here to upload a new file
              </p>
            </div>
            <label
              class="label-file-widget-input"
            >
              Choose a file
            </label>
            <input
              autocomplete="off"
              id="field-my-field"
              multiple=""
              name="my-field"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
          </div>
          <div
            class="field-file-name"
          />
        </div>
      </div>
    </div>
  </div>
  ,
</div>
`;
